packages/kit/src/runtime/client/renderer.js

```diff
@@ -84,11 +84,10 @@ export class Renderer {
 
 		/** @param {MouseEvent} event */
 		const trigger_prefetch = (event) => {
-			/** @type {HTMLAnchorElement | SVGAElement} */
+			if (!(event.target instanceof Node)) return;
```

Comment:
i understand why this is here, but it means we're shipping completely unnecessary code to clients just to appease TypeScript. is there another way?

Reply:
It's easy enough to type case, but I was worried this might be necessary. I'm not familiar enough with the intricacies, but I would expect that if it were unnecessary then `event.target` would have the type of `Node` rather than just `EventTarget`. Otherwise I'm not sure why they'd have made that choice. I was worried you might be able to mouse over the window or something that's not a `Node`

Reply:
The event will fire if you mouseover `<html>` (if the mouse moves from out of the window into the window) or `<body>` (if there's nothing else under the mouse), and neither `HTMLHtmlElement` or `HTMLBodyElement` is an instance of `Node`. Both will work just fine with `find_anchor` though. I don't totally understand the Venn diagram that includes `Node` and `HTMLElement` and the various `SVGElement`s